modules/040-node-manager/hooks/clear_nodeuser_errors.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// this hook figure out minimal ingress controller version at the beginning and on IngressNginxController creation
+// this version is used on requirements check on Deckhouse update
+// Deckhouse would not update minor version before pod is ready, so this hook will execute at least once (on sync)
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube/object_patch"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+
+	nodeuserv1 "github.com/deckhouse/deckhouse/modules/040-node-manager/hooks/internal/v1"
+)
+
+const (
+	nodeForClearSnapName     = "nodes_for_clear"
+	nodeUserForClearSnapName = "nodeuser_for_clear"
+)
+
+type nodeUsersForClear struct {
+	Name         string
+	StatusErrors map[string]string
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/node-manager",
+	Kubernetes: []go_hook.KubernetesConfig{
```

Comment:
Looks like this hook should be schedule periodically as this hook https://github.com/deckhouse/deckhouse/blob/29d02b2f3fc8565e25142ff55241b25ed4e4cab0/modules/005-external-module-manager/hooks/cleanup_releases.go#L55